File: widgets/src/iosMain/swift/ProgressbarWidgetViewFactory.swift
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */

import UIKit
import MultiPlatformLibrary

public class ProgressBarWidgetViewFactory {
  
  public static func setup() {
    // TODO такое вот можно легко в котлине реализовать, чтобы не требовался свифт файл дополнительный и без вызова setup обойтись можно будет
    ProgressBarWidgetViewFactoryKt.progressBarWidgetViewFactory = { (controller, widget) -> UIView in
      let indicator = UIActivityIndicatorView(style: .whiteLarge) // TODO пробрасывать через стиль виджета
      indicator.color = UIColor.gray
      indicator.translatesAutoresizingMaskIntoConstraints = false
      indicator.startAnimating()
      return indicator
    }
  }
}

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/progressBarWidgetViewFactory.kt
/*
 * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
 */

package dev.icerock.moko.widgets

import dev.icerock.moko.graphics.toUIColor
import dev.icerock.moko.widgets.core.VFC
import platform.UIKit.UIActivityIndicatorView
import platform.UIKit.UIActivityIndicatorViewStyleWhite
import platform.UIKit.UIColor
import platform.UIKit.translatesAutoresizingMaskIntoConstraints

actual var progressBarWidgetViewFactory: VFC<ProgressBarWidget> = { viewController, widget ->
    val style = widget.style

    // TODO add change of size style
    UIActivityIndicatorView(activityIndicatorStyle = UIActivityIndicatorViewStyleWhite).apply {
        translatesAutoresizingMaskIntoConstraints = false
        color = style.color?.toUIColor() ?: UIColor.darkGrayColor
        startAnimating()
    }
}
